crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -108,28 +108,62 @@ pub(crate) enum EnumVariantFields<'a> {
     Unit,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum ReflectType {
```

Comment:
Nit: Could you add doc comments, please? `ReflectType` is a bit vague on its own.